url8c14fd53ee_luca89_itaco_tgz-pJ8-main_RunnerJ8/src/compilatore/CTarget.java

```diff
@@ -0,0 +1,543 @@
+			throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		parteSinistra.scriviCodice(this);
+		ps.print(" > ");
+		parteDestra.scriviCodice(this);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#minore(compilatore.Espressione,
+	 * compilatore.Espressione)
+	 */
+	@Override
+	public void minore(Espressione parteSinistra, Espressione parteDestra)
+			throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		parteSinistra.scriviCodice(this);
+		ps.print(" < ");
+		parteDestra.scriviCodice(this);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#uguaglianza(compilatore.Espressione,
+	 * compilatore.Espressione)
+	 */
+	@Override
+	public void uguaglianza(Espressione parteSinistra, Espressione parteDestra)
+			throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		parteSinistra.scriviCodice(this);
+		ps.print(" == ");
+		parteDestra.scriviCodice(this);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * compilatore.ScrittoreTarget#espressioneInParentesi(compilatore.Espressione
+	 * )
+	 */
+	@Override
+	public void espressioneInParentesi(Espressione ex)
+			throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("(");
+		ex.scriviCodice(this);
+		ps.print(")");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#stampa(compilatore.Espressione)
+	 */
+	@Override
+	public void stampa(Espressione espressione) throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("printf(\"%d\",");
+		espressione.scriviCodice(this);
+		ps.print(");\n");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#stampa(java.lang.String)
+	 */
+	@Override
+	public void stampa(String stringa) throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("printf(");
+		costante(stringa);
+		ps.print(");\n");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#leggi(java.lang.String)
+	 */
+	@Override
+	public void leggi(String identificatore) throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("scanf(\"%d\", &" + identificatore + ");\n");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#leggiElementoVettore(java.lang.String,
+	 * compilatore.Espressione)
+	 */
+	@Override
+	public void leggiElementoVettore(String identificatore, Espressione indice)
+			throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("scanf(\"%d\", " + identificatore + "+");
+		indice.scriviCodice(this);
+		ps.print(");\n");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see compilatore.ScrittoreTarget#se(compilatore.Espressione,
+	 * compilatore.Blocco)
+	 */
+	@Override
+	public void se(Espressione ex, Blocco codice) throws EccezioneSemantica {
+		// TODO Auto-generated method stub
+		ps.print("if(");
+		ex.scriviCodice(this);
+		ps.print("){\n");
+		codice.scriviCodice(this);
+		ps.print("\n}\n");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 287:291, 319:323.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.